code/PlayerInfo.cpp

```diff
@@ -1342,6 +1345,78 @@ void PlayerInfo::UnloadCargo()
 
 
 
+
+void PlayerInfo::Refuel()
+{
+  double price = planet->GetFuelPrice();
+  // Refuels only if the price is in the preferred range or 0 or the player is overriding
+  // the preferences using the "Refuel All"-button.
+  bool rechargesFuel = (price < Preferences::GetMaxPrice() || !price) && price >= 0;
+  int rechargedFuel = 0;
+  bool hasSpaceport = planet->HasSpaceport() && planet->CanUseServices();
+  for(const shared_ptr<Ship> &ship : ships)
+    if(!ship->IsDisabled())
+    {
+      if(ship->GetSystem() == system)
+      {
+        ship->Recharge(hasSpaceport);
+        if(rechargesFuel)
+          rechargedFuel += ship->Refuel(1);
+        ship->Cargo().TransferAll(cargo);
+        ship->SetPlanet(planet);
+      }
+      else
+        ship->Recharge(false);
+    }
+  
+  // Pays the fuel and messages the price that was paid.
+  if(rechargedFuel && price > 0)
+  {
+    int fuelPrice = rechargedFuel*price;
+    ostringstream out;
+    out << "You paid " << fuelPrice << " credits to buy " << rechargedFuel << " units of fuel.";
+    Messages::Add(out.str());
```

Comment:
As you can see here there should be a message. Have you changed the messaging system in the last weeks?